design/volume-snapshot-data-movement/volume-snapshot-data-movement.md

```diff
@@ -0,0 +1,763 @@
+- Data Persistency and Management: Persistent the data in backup storage and manage its security, redundancy, accessibility, etc. through backup repository. This has been covered by the [Unified Repository design][1]
+
+At present, Velero supports moving file system data from PVs through Pod Volume Backup (a.k.a. file system backup). However, it backs up the data from the live file system, so it should be the last option when more consistent data movement (i.e., moving data from snapshot) is not available.  
+
+Moreover, we would like to create a general workflow to variations during the data movement, e.g., data movement plugins, different snapshot types, different snapshot accesses and different data accesses. 
+
+## Goals
+
+- Create components and workflows for Velero to move data based on volume snapshots
+- Create components and workflows for Velero built-in data mover
+- Create the mechanism to support data mover plugins from third parties
+- Implement CSI snapshot data movement on file system level
+- Support different data accesses, i.e., file system level and block level
+- Support different snapshot types, i.e., CSI snapshot, volume snapshot API from storage vendors
+- Support different snapshot accesses, i.e., through PV generated from snapshots, and through direct access API from storage vendors
+- Reuse the existing Velero generic data path as creatd in [Unified Repository design][1]
+
+## Non-Goals
+
+- The implementation details of block level access will be included in a future design
+- Most of the components are generic, but the Exposer is snapshot type specific or snapshot access specific. The current design covers the implementation details for exposing CSI snapshot to host path access only, for other types or accesses, we may need a separate design
+- The current workflow focuses on snapshot-based data movements. For some application/SaaS level data sources, snapshots may not be taken explicitly. We don’t take them into consideration, though we believe that some workflows or components may still be reusable.
+
+## Architecture of Volume Snapshot Data Movement
+
+## Workflows  
+
+Here are the diagrams that illustrate components and workflows for backup and restore respectively.  
+For backup, we intend to create an extensive architecture for various snapshot types, snapshot accesses and various data accesses. For example, the snapshot specific operations are isolated in Data Mover Plugin and Exposer. In this way, we only need to change the two modules for variations. Likely, the data access details are isolated into uploaders, so different uploaders could be plugged into the workflow seamlessly.  
+
+For restore, we intend to create a generic workflow that could for all backups. This means the restore is backup source independent. Therefore, for example, we can restore a CSI snapshot backup to another cluster with no CSI facilities or a CSI driver the same as the source cluster.  
```

Comment:
`we can restore a CSI snapshot backup to another cluster with no CSI facilities or a CSI driver **the same as** the source cluster.`

`the same as` part is bit ambiguity to me. In the context, should it be `different from`?

Reply:
Yes, corrected it. Thanks.